### ISSUE TITLE
better lines deleting

### ISSUE THREAD

**Issue by craynic** [geany/geany-plugins#782]
Using `ctrl+v d` will leave a blank line, cause it inconvenient to paste. It is slightly different from the same operation in vim. Hope it could be the same.

**Comment by elextr** [comment]
The cursor is put after a paste.  If your paste includes the end of line characters then the cursor will be on another line, and thats what gets duplicated.

**Comment by techee** [comment]
This is unfortunately one of the hard-to-fix problems because of how the source code editing component (Scintilla) works compared to vim. In vim EOL behaves like a normal character at the end of line while in Scintilla just breaks the line but the EOL character isn't at the end of the line.

**Comment by elextr** [comment]
@techee not sure what you mean, the EOLs are definitely in the Scintilla buffer like all the other characters.

**Comment by techee** [comment]
Yes, but you cannot put cursor on the EOL character (which is possible in Vim) - the cursor is just in front of the EOL character and this is why copying a line is off-by-1 and excludes the EOL character in the plugin.

**Comment by elextr** [comment]
Ahh, its an SBC problem (stupid block cursor :) thats not really a Scintilla long suit, IIRC its not maintained by Neil.
